### PR TITLE
feat: Create Github Action

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Created Github Action for use by other repos. 

# Changelog

- Add action.yml
  - This defines how the Github Action is used by other repos
  - Requires an `ot3-firmware-commit-id` parameter to tell the action where to pull the source for the firmware from
- Update .gitignore to ignore ot3-firmware directory that is pulled in as part of build
- Dockerfile Changes
  - Install prebuilt binary for cmake instead of building from source
  - Add `firmware` build stage
  - Make the apt stuff quieter
- Rename run.sh to build_system.sh
  - Add pulling firmware
  - Add pulling firmware from specific commit
  - Add running headless
- Remove sending stderr to /dev/null in setup_can.sh (It was masking errors) 

# Review requests

None

# Risk assessment

None

### CODE REVIEW

**Comment by DerekMaggio** [comment]
I need to create a `v0.1` release after this is merged for other repos to use

File: action.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: 'OT-3 Emulator'
+description: 'Spin up emulator with specific version of firmware code'
+inputs:
+  ot3-firmware-commit-id:
```
Comment (sotaoverride) [inline action.yaml:4]: where does this get assigned a value?

Comment (DerekMaggio) [reply action.yaml:4]: It gets assigned a value when you run the Github Action.
See [running the action in ot-3 firmware](https://github.com/Opentrons/ot3-firmware/blob/ot3-emulator-action/.github/workflows/run_emulation.yaml) for an example

Comment (DerekMaggio) [reply action.yaml:4]: I will be updating the README with a link this example when I get it merged over in `ot3-firmware`

**Comment by DerekMaggio** [comment]
Moving this over to another pull request that is linked to JIRA ticket

**Comment by DerekMaggio** [comment]
Moved to https://github.com/Opentrons/ot3-emulator/pull/11

**Comment by DerekMaggio** [comment]
Moved to https://github.com/Opentrons/ot3-emulator/pull/11

**Pull request closed without merging**